Code/Framework/AzFramework/AzFramework/Components/NativeUISystemComponent.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzFramework/Application/Application.h>
+#include <AzFramework/Input/Devices/Gamepad/InputDeviceGamepad.h>
+#include <AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard.h>
+#include <AzFramework/Input/Devices/Motion/InputDeviceMotion.h>
+#include <AzFramework/Input/Devices/Mouse/InputDeviceMouse.h>
+#include <AzFramework/Input/Devices/Touch/InputDeviceTouch.h>
+#include <AzFramework/Input/Devices/VirtualKeyboard/InputDeviceVirtualKeyboard.h>
+#include <AzFramework/Windowing/NativeWindow.h>
+
+namespace AzFramework
+{
+    //! This system will take care of reflecting physics material classes and
+    //! registering the physics material asset.
+    class NativeUISystemComponent
+        : public AZ::Component
+    {
+    public:
+        AZ_COMPONENT(AzFramework::NativeUISystemComponent, "{F2BA607F-08A9-42DD-9C0D-F324BFDD4DFD}");
```

Comment:
The qualification should be removed from the reflection.
This should just be NativeUiSystemComponent.
The fully qualified name makes it complicated to access in Scripting languages as it would require sanitizing the name to deal with the `::`
```suggestion
        AZ_COMPONENT(NativeUISystemComponent, "{F2BA607F-08A9-42DD-9C0D-F324BFDD4DFD}");
```